internal/compose/compose.go

```diff
@@ -68,17 +82,17 @@ func (p *portMapping) UnmarshalYAML(node *yaml.Node) error {
 
 		var s struct {
 			HostIP    string `yaml:"host_ip"`
-			Target    int
-			Published int
+			Target    intOrStringYaml
```

Comment:
If we start observing such problems, then I guess we'll end up soon with code switches depending on v1/v2.  Let's drop a comment here, why do we need to handle it this way.